src/include/network/services/host.rb

```diff
@@ -253,12 +270,9 @@ def HostsMainDialog(standalone)
           Host.clear
 
           table_items.each do |row|
-            encoded_aliases = Punycode.EncodePunycodes([row[3] || ""])
-            encoded_canonical = Punycode.EncodeDomainName(row[2] || "")
-            value = encoded_canonical + " " + encoded_aliases.join(" ")
-            key = row[1] || ""
+            hostnames = encode_hosts_line(row[2], (row[3] || "").split)
```

Comment:
I still finding it difficult to understand.  I am going to come back with my previous suggestion 

```ruby

table_items.each do |key = '', canonical = '', aliases = [] |
  hostnames = encoded_host_line(canonical, aliases)

  Host.add_name(key, )
end
```

What do you think?

Reply:
i don't think that this would work without further intervention in the code (```table_items``` is an array of something what used to be Term ```Item``` in YCP times). I think I was intrusive enough for P1 hotfix ;-)